redis/runtime/src/main/java/io/quarkiverse/langchain4j/redis/RedisEmbeddingStore.java

```diff
@@ -0,0 +1,246 @@
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public RedisEmbeddingStore(ReactiveRedisDataSource ds, RedisSchema schema) {
+        this.ds = ds;
+        this.schema = schema;
+        try {
+            createIndexIfDoesNotExist();
+        } catch (Exception e) {
+            LOG.error(e);
+        }
+    }
+
+    private void createIndexIfDoesNotExist() {
+        List<String> indexes = ds.search().ft_list().await().indefinitely();
+        if (!indexes.contains(schema.getIndexName())) {
+            // TODO: rewrite to use the typesafe data source API
+            Request request = Request.cmd(Command.FT_CREATE)
+                    .arg(schema.getIndexName())
+                    .arg("ON")
+                    .arg("JSON")
+                    .arg("PREFIX")
+                    .arg("1")
+                    .arg(schema.getPrefix())
+                    .arg("SCHEMA");
+            schema.defineFields(request);
+            LOG.debug(
+                    "Creating index with command: " + request.toString().replaceAll("\r\n", " "));
+            ds.getRedis().send(request).await().indefinitely();
+        } else {
+            LOG.debug("Index in Redis already exists: " + schema.getIndexName());
+        }
+    }
+
+    @Override
+    public String add(Embedding embedding) {
+        String id = randomUUID();
+        add(id, embedding);
+        return id;
+    }
+
+    @Override
+    public void add(String id, Embedding embedding) {
+        addInternal(id, embedding, null);
+    }
+
+    @Override
+    public String add(Embedding embedding, TextSegment textSegment) {
+        String id = randomUUID();
+        addInternal(id, embedding, textSegment);
+        return id;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(singletonList(id), singletonList(embedding), embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (ids.isEmpty() || ids.size() != embeddings.size() || (embedded != null && embedded.size() != embeddings.size())) {
+            throw new IllegalArgumentException("ids, embeddings and embedded must be non-empty and of the same size");
+        }
+        ds.withTransaction(ds -> {
```

Comment:
Why do you need a transaction?

Reply:
(Redis transactions are not relational DB transaction)

Reply:
I'll remove the tx